qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/AppointmentSteps.java

```diff
@@ -0,0 +1,240 @@
+		dashboardPage = findPatientPage.clickOnFirstPatient();
+	}
+
+	@After(RunTest.HOOK.SELENIUM_APPOINTMENT)
+	public void destroy() {
+		deletePatient(testPatient);
+		quit();
+	}
+
+	@Given("a user clicks on Request Appointment link from the Patient dashboard")
+	public void loadRequestAppointmentPage() {
+		requestAppointmentPage = (RequestAppointmentPage) dashboardPage.clickOnRequest();
+	}
+
+	@When("the system loads Request Appointment page")
+	public void systemLoadsRequestAppointmentPage() {
+		assertTrue(textExists("Request Appointment"));
+	}
+
+	@And("a user fills in the appointment request details")
+	public void captureAppointmentRequestDetails() {
+		requestAppointmentPage.enterAppointmentType("Oncology");
+		requestAppointmentPage.enterMinimumValue("0");
+		requestAppointmentPage.selectMinimumUnits("Day(s)");
+		requestAppointmentPage.enterMaximumValue("2");
+		requestAppointmentPage.selectMaximumUnits("Day(s)");
+		requestAppointmentPage.addAppointmentRequestNote("I need to see a doctor first");
+	}
+
+	@And("a user clicks save button")
+	public void saveAppointmentRequest() {
+		dashboardPage = requestAppointmentPage.saveRequest();
+	}
+
+	@Then("the system adds appointment request into the appointment requests table")
+	public void systemSavesAppointmentRequest() {
+		assertNotNull(dashboardPage.getAppointmentRequestsList());
+	}
+
+	@When("a user clicks on Schedule Appointment link from the Patient dashboard")
+	public void loadManageAppointmentsPage() {
+		manageAppointmentsPage = (ManageAppointmentsPage) dashboardPage.goToManageAppointments();
+	}
+
+	@And("the system loads Manage Appointments page")
+	public void systemLoadsManageAppointmentsPage() {
+		assertTrue(manageAppointmentsPage.containsText(testPatient.identifier));
+	}
+
+	@And("a user clicks on Cancel this Request icon")
+	public void deleteAppointmentRequest() {
+		manageAppointmentsPage.deleteRequest() ;
+	}
+
+	@Then("the system deletes appointment request from the request table")
+	public void systemDeletesAppointmentRequest() {
+		assertTrue(manageAppointmentsPage.containsText("No appointment requests"));
+	}
+
+	@When("a user clicks Appointment scheduling app from the home page")
+	public void loadAppointmentSchedulingPage() {
+		dashboardPage.goToHomePage();
+		appointmentSchedulingPage = (AppointmentSchedulingPage ) homePage.goToAppointmentScheduling();
+	}
+
+	@Then("the system loads the appointment scheduling page")
+	public void systemLoadsAppointmentSchedulingPage() {
+		assertTrue(textExists("Appointment Scheduling"));
+	}
+
+	@And("a user clicks on Manage Provider Schedules app")
+	public void loadManageProviderSchedulesPage() {
+		manageProviderSchedulesPage = (ManageProviderSchedulesPage) appointmentSchedulingPage.goToManageProviderSchedules();
+	}
+
+	@And("the system loads Manage Provider Schedules page")
+	public void systemLoadsManageProviderSchedulesPage() {
+		assertTrue(textExists("Manage Appointment Blocks"));
```

Comment:
Is this the manager provider schedules or manage appointment blocks page?

Reply:
@dkayiwa it is  manage provider schedules  page. However, this page together with  appointment blocks page are both loaded at the same url "/appointmentschedulingui/scheduleProviders.page"  and both contain "Manage Appointment Blocks" on display. It is only they implement different functionalities.